src/bugreporter/odemis_bugreporter.py

```diff
@@ -739,6 +789,74 @@ def open_failed_upload_dlg(self, error: str):
             webbrowser.open('https://support.delmic.com/open.php')
 
 
+    def _add_attachment_dialog(self, evt: wx.Event) -> List[str]:
+        """Open a dialog to select files to attach to the bug report"""
+
+        dlg = wx.FileDialog(self, message="Select files to attach",
+                            wildcard="All files (*.*)|*.*",
+                            style=wx.FD_OPEN | wx.FD_MULTIPLE)
+
+        if dlg.ShowModal() == wx.ID_OK:
+            paths = dlg.GetPaths()
+        else:
+            paths = []
+        dlg.Destroy()
+
+        # Add the selected files to the list of attachments
+        self.bugreporter.user_attachments.extend(paths)
+        logging.info("Added attachments: %s", paths)
+
+        # update label
+        self._update_attachment_label()
+
+    def _remove_attachments(self, evt: wx.Event) -> None:
+        """Remove all user-selected attachments"""
+        self.bugreporter.user_attachments.clear()
+        logging.info("Removed all attachments")
+
+        # update label
+        self._update_attachment_label()
+
+    def _update_attachment_label(self) -> None:
+        """Update the label showing the number of attached files"""
+
+        paths = self.bugreporter.user_attachments
+
+        # no files attached
+        if len(paths) == 0:
+            self.attached_files_lbl.SetForegroundColour(wx.Colour(255, 255, 255))
+            self.attached_files_lbl.SetLabel("No files attached")
+            return
+
+        valid_attachments = _validate_user_attachment_size(self.bugreporter.user_attachments)
+        self.report_btn.Enable(valid_attachments)
+
+        if not valid_attachments:
+            max_str = int(MAX_ATTACHMENT_SIZE / (1024*1024))
+            self.attached_files_lbl.SetLabel(f"Total size of attachments exceeds the limit ({max_str}MB). Please remove some attached files.")
+            # set text color to red
+            self.attached_files_lbl.SetForegroundColour(wx.Colour(255, 0, 0))
+            self.Layout()
+            return
+
+        # show the number of attached files
+        if len(paths) == 1:
+            label = "1 file attached"
+        else:
+            label = f"{len(paths)} files attached"
+
+        # resize the label to fit the text
+        self.attached_files_lbl.Wrap(self.attached_files_lbl.GetSize().width)
+        self.Layout()
```

Comment:
In my tests, this didn't work. The file names overlap with the "small text". In order to get it to work, I had to:
* move it to the very end of this function
* change it to `self.panel.Layout()`

I think you actually need to use `self.panel.Layout()` everywhere you currently use `self.Layout()`.